lib/puppet/type/notify.rb

```diff
@@ -6,7 +6,7 @@ module Puppet
   Type.newtype(:notify) do
     @doc = "Sends an arbitrary message to the agent run-time log."
 
-    newproperty(:message) do
+    newproperty(:message, :idempotent => false) do
```

Comment:
If we do a whitelist then that's probably going to have to live in core; with an idempotent property that can be done on a per-custom type basis which is nice. Also, this implementation works (it works, right?) and we're under pressure to ship, so I would rather just take this as-is and roll with it. If this is YAGNI then it's very small YAGNI with limited scope and minimal impact.